### DISCUSSION TITLE
Colab Notebook for Training Your Personal Language Models from your own ChatGPT Discussions

### DISCUSSION THREAD

**Discussion by innermost47** [oobabooga/text-generation-webui#2001] [Show and tell]
Hey everyone!

I've created a Google Colab notebook that helps to generate a training dataset in JSON format, specifically designed for training your own language models.

It uses our personal conversations downloaded from ChatGPT as reference data. By leveraging these conversational data points, it can create a highly interactive and conversational dataset, something that can be challenging to achieve otherwise.

Once run, the script identifies instructions and their corresponding outputs from the conversations, structures these into individual JSON objects, and saves them into a JSON file. This creates a ready-to-use dataset that you can directly plug into your local language model training regime.

The goal here is to streamline the process of creating high-quality, conversational datasets and make language model training more accessible and practical for everyone.

You can find the notebook here: https://colab.research.google.com/drive/1ZZnEM-9_fIuSQ6X-kEoM_9jUmtYCT32i?usp=sharing

Give it a try and let me know what you think! I'm eager to hear your feedback and make improvements based on your experiences. Happy coding!

**Comment by spreck** [comment]
This looks great! Could you expand on "you can directly plug into your local language model training regime" perhaps by outlining an example? Is this something we could do on colab as well to make a LORA?
